packages/fxa-payments-server/src/lib/test-utils.tsx

```diff
@@ -513,6 +513,7 @@ export const MOCK_ACTIVE_SUBSCRIPTIONS_AFTER_SUBSCRIPTION = [
 export const MOCK_CUSTOMER = {
   billing_name: 'Jane Doe',
   payment_type: 'card',
+  payment_provider: 'stripe',
```

Comment:
If I _don't_ modify `MOCK_CUSTOMER` to have the `payment_provider` key with a value of `"stripe"`, the `PaymentForm` test 'with existing card' > 'calls the submit handler' test case fails, but it passes if I _do_ modify it. I still don't understand why... Would you know? Both new and existing Stripe customer `PaymentForm`s have a submit button with a `data-testid="submit"`.

I do know that when `payment_provider` is `undefined`, we show the new customer version of `SubscriptionCreate`. When it's `"stripe"` we show the returning Stripe customer version of `SubscriptionCreate`.

Reply:
Now that we are checking the 'payment_provider' value and not `last4`, so like you said, we show the new customer form, and you cannot submit that form unless you enter the valid values into the fields.

Reply:
Hm... true, but isn't that also true for returning Stripe customers? I.e. they must first check the `Checkbox` before they can submit the form?

Reply:
> 
> 
> Hm... true, but isn't that also true for returning Stripe customers? I.e. they must first check the `Checkbox` before they can submit the form?

Very true.  I don't have the answer off-hand.  I suspect it has something to do with the test setup.

Reply:
Yeah I know there's a `confirm` prop that can be passed it that looks to bypasses disabling the submit button until all form elements have been filled in with valid data/completed; I just didn't see that it was being set to `true` anywhere.